packages/sdk/src/referral.ts

```diff
@@ -687,4 +656,141 @@ export class ReferralProvider {
 
     return vtTxs.flat();
   }
+
+  public async claimPartially({
+    payerPubKey,
+    referralAccountPubKey,
+    strategy,
+    withdrawalableTokenAddress,
+  }: ClaimPartiallyVariable): Promise<VersionedTransaction[]> {
+    const blockhash = (await this.connection.getLatestBlockhash()).blockhash;
+    const lookupTableAccount = await this.connection
+      .getAddressLookupTable(
+        new PublicKey("GBzQG2iFrPwXjGtCnwNt9S5eHd8xAR8jUMt3QDJpnjud"),
+      )
+      .then((res) => res.value);
+
+    const referralAccount = await this.program.account.referralAccount.fetch(
+      referralAccountPubKey,
+    );
+    const project = await this.program.account.project.fetch(
+      referralAccount.project,
+    );
+    const projectAuthority = this.getProjectAuthorityPubKey(project);
+
+    const result = await this.connection.getMultipleAccountsInfo(
+      withdrawalableTokenAddress,
+    );
+
+    const claimInstructionParams = await Promise.all(
+      result.map(async (item) => {
+        const tokenAccountData = AccountLayout.decode(item.data);
+
+        const isToken2022 = item.owner.equals(TOKEN_2022_PROGRAM_ID);
+        const tokenProgramId = isToken2022
+          ? TOKEN_2022_PROGRAM_ID
+          : TOKEN_PROGRAM_ID;
```

Comment:
tokenProgramId is the item.owner, no need to check here, just pass in item.owner will do

Reply:
oh ya.